lib/auth/keystore/keystore.go

```diff
@@ -128,6 +129,9 @@ func NewKeyStore(cfg Config) (KeyStore, error) {
 	if cfg.Path == "" {
 		return NewRawKeyStore(&RawConfig{cfg.RSAKeyPairSource}), nil
 	}
+	if !modules.GetModules().Features().HSM {
+		return nil, trace.AccessDenied("HSM support is only available with an enterprise license")
+	}
```

Comment:
Not sure if this remark is relevant but wonder about backward compatibility, For instance is client already uses HSM and start the cluster update from upgrading the proxy version.

Reply:
Luckily this is not relevant because we have not released the feature yet